### ISSUE TITLE
Escaping + triggers selector-combinator-space-before

### ISSUE THREAD

**Issue by KeironLowe92** [stylelint/stylelint#1407]
You can escape + symbols so that they can be used as class names, but this triggers the `selector-combinator-space-before` rule.

Rule: `selector-combinator-space-before`
Selector: `.u-mb\+ { margin-bottom: 40px; }`

Config: 
```
{
    "rules": {
        "selector-combinator-space-before": "always"
    }
}
````

Version: `6.5.1`
Running: `Node API` via gulp-stylelint
Does your issue relate to non-standard syntax: No
What did you expect to happen?: "No warnings to be flagged."
What actually happened (e.g. what warnings or errors you are getting)?:
```
27:11: Expected single space before "+" combinator - selector-combinator-space-before
```

**Comment by evilebottnawi** [comment]
@KeironLowe92 Thank for feed back, be good if you PR failed tests.
